src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -1680,7 +1680,7 @@ public static boolean isRegionProtectedFromBuildPostCache(final Player player, f
 					final Kingdom kingdom = land.getKingdom();
 					if (kPlayer.isAdmin()
 							|| (!protectDuringInvasions && !land.getInvasions().isEmpty() && land.getInvasions().values().stream().anyMatch(i -> i.getInvader().equals(kPlayer))) // Protection during invasions is off, and player is currently invading; allow
-							|| (land.getStructure() != null && land.getStructure() instanceof Regulator && ((Regulator) land.getStructure()).hasAttribute(player, Attribute.BUILD))) { // There is a regulator on site which allows the player to build; allow
+							|| (land.getStructures() != null && land.getStructures() instanceof Regulator && ((Regulator) land.getStructures()).hasAttribute(player, Attribute.BUILD))) { // There is a regulator on site which allows the player to build; allow

```

Comment:
Land#getStructures returns a Map<SimpleLocation, Structure>, and it is never null. We can remove the null check, but we'll have to do `land.getStructures().values().stream().anyMatch(s -> s instanceof Regulator && ((Regulator) s).hasAttribute(player, Attribute.BUILD));` (I think).